cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -157,7 +154,26 @@ func (g *genDeepCopy) copyableWithinPackage(t *types.Type) bool {
 		return false
 	}
 	// TODO: Consider generating functions for other kinds too.
-	return t.Kind == types.Struct
+	if t.Kind != types.Struct {
+		return false
+	}
+	// TODO: This should be removed once we start generating public DeepCopy
+	// functions per directory.
+	if t.Name.Package != g.targetPackage {
+		// We won't be able to access private fields.
+		// Thus, this type cannot have private fields.
+		for _, member := range t.Members {
+			if strings.ToLower(member.Name[:1]) == member.Name[:1] {
+				return false
+			}
+			// TODO: This is a temporary hack, to make avoid generating function
+			// for conversion.Equalities. We should get rid of it.
```

Comment:
No - actually I meant this one:
https://github.com/kubernetes/kubernetes/blob/master/pkg/conversion/deep_equal.go#L25

Basically, generator is currently not able to generate the deep-copy function for such structs. Since this is actually not needed for now and currently it's not causing us any other problems (all functions that are supposed to be generated are generated), I think it's temporary fine.